### PR TITLE
MTDSA-14139: Added TYS code for Delete other expenses

### PR DESCRIPTION
**Pull request by LouiseWong98**

### CODE REVIEW

File: app/v1/services/DeleteOtherExpensesService.scala [hunk 1]
```diff
@@ -40,19 +39,23 @@ class DeleteOtherExpensesService @Inject() (deleteOtherExpensesConnector: Delete
...
 
-  private def desErrorMap: Map[String, MtdError] =
-    Map(
+  private val downstreamErrorMap: Map[String, MtdError] = {
+    val errors = Map(
       "INVALID_TAXABLE_ENTITY_ID" -> NinoFormatError,
-      "FORMAT_TAX_YEAR"           -> TaxYearFormatError,
+      "INVALID_TAX_YEAR"          -> TaxYearFormatError,
       "NO_DATA_FOUND"             -> NotFoundError,
+      "INVALID_CORRELATIONID"     -> StandardDownstreamError,
```
Comment (tom-lloyd-hmrc) [inline app/v1/services/DeleteOtherExpensesService.scala:50]: I think this needs to be added to the Tech Spec too.

File: test/v1/services/DeleteOtherExpensesServiceSpec.scala [hunk 2]
```diff
@@ -54,25 +55,30 @@ class DeleteOtherExpensesServiceSpec extends ServiceSpec {
...
+            .returns(Future.successful(Left(ResponseWrapper(correlationId, DownstreamErrors.single(DownstreamErrorCode(downstreamErrorCode))))))
 
           await(service.deleteOtherExpenses(requestData)) shouldBe Left(ErrorWrapper(correlationId, error))
         }
 
-      val input = Seq(
+      val errors = Seq(
         "INVALID_TAXABLE_ENTITY_ID" -> NinoFormatError,
-        "FORMAT_TAX_YEAR"           -> TaxYearFormatError,
+        "INVALID_TAX_YEAR"          -> TaxYearFormatError,
```
Comment (tom-lloyd-hmrc) [inline test/v1/services/DeleteOtherExpensesServiceSpec.scala:70]: I think INVALID_CORRELATIONID needs to be added here too.

File: app/v1/services/DeleteOtherExpensesService.scala (hunk 1 above)
Comment (LouiseWong98) [reply app/v1/services/DeleteOtherExpensesService.scala:50]: This is in the old tech spec already but somehow it wasnt added into the code, so I added back in

Comment (tom-lloyd-hmrc) [reply app/v1/services/DeleteOtherExpensesService.scala:50]: Sorry yeah I was looking at the new one but it shouldn't be in there 🤦

**Review by tom-lloyd-hmrc — APPROVED** [review]

**Pull request merged**